content/articles/build-nextjs-with-mongodb-and-deploy-on-vercel/index.md

```diff
@@ -0,0 +1,982 @@
+
+### Goal
+
+In this guide, you will learn how to create a Next.js application using MongoDB and then deploy the application on [Vercel](https://vercel.com/dashboard). We will create a simple blog application that communicates with MongoDB. This will allow you to add a new post to the blog app and update or delete a post from the app.
+
+### Prerequisites
+
+To follow along with this guide, the following requirements will help you navigate around this application and run it without a big hassle.
+
+- [Node.js](https://nodejs.org/en/) installed. Node.js will help us run NPM commands to install any necessary dependencies that will help us make a Next.js application and communicate with a MongoDB database.
+- Since we are working with [Next.js](https://nextjs.org/), prior knowledge of how to use [Next.js](https://www.youtube.com/watch?v=mTz0GXj8NN0&t=53s) will be helpful.
+- We are using [MongoDB](https://www.mongodb.com/) as the database for the post storage. Prior knowledge of working with this database will be of great importance.
+
+### Table of content
+
+- [Goal](#goal)
+- [Prerequisites](#prerequisites)
+- [Table of content](#table-of-content)
+- [Installing Next.js](#installing-nextjs)
+- [Laying out the postcards and a Navbar](#laying-out-the-postcards-and-a-navbar)
+- [Setting up Next.js pages](#setting-up-nextjs-pages)
+- [Setting up MongoDB using MongoDB Atlas](#setting-up-mongodb-using-mongodb-atlas)
+- [Connecting to MongoDB Atlas cluster](#connecting-to-mongodb-atlas-cluster)
+- [Setting up the API route](#setting-up-the-api-route)
+  - [Add handler function](#add-handler-function)
+  - [`components/PostCard.js`](#componentspostcardjs)
+  - [`pages/add-post.js`](#pagesadd-postjs)
+  - [`pages/index.js`](#pagesindexjs)
+  - [Fetching posts](#fetching-posts)
+  - [Adding a post](#adding-a-post)
+  - [Updating a post](#updating-a-post)
+  - [Deleting a post](#deleting-a-post)
+- [Hosting to Vercel](#hosting-to-vercel)
+- [Conclusion](#conclusion)
+
+### Installing Next.js
+
+Next.js provides a one-time command that helps developers to scaffold a Next.js project. This will fasten the development time and serve a developer a well already structured application. First, create a folder and change the directory to it. Then, to create a Next.js app, run the command `npx create-next-app` inside that directory. This will then prompt you with what name you want to call your app. As a shortcut, you can still run this one command with `npx create-next-app` followed the app's name, as shown below.
```

Comment:
```suggestion
Next.js provides a one-time command that helps developers to scaffold a Next.js project. This will fasten the development time and serve a developer a well already structured application. First, create a folder and change the directory to it. Then, to create a Next.js app, run the command `npx create-next-app` inside that directory. This will then prompt you with what name you want to call your app. As a shortcut, you can still run this one command with `npx create-next-app` followed the app's name, as shown below:
```